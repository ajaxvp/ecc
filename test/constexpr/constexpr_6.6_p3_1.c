/* ISO: 6.6 (3); invalid operators for constant expressions in evaluated sections */

int a;

// assignment operator
int b = a = 5;

// increment operators
int c = a += 1;
int d = a++;

// decrement operators
int e = a -= 1;
int f = a--;

// function call operator

static int g(void) {}

int h = g();

// comma operators
int i = (f, h);
