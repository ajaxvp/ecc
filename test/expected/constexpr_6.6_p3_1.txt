ecc: error: [6:9] in static initialization: assignment operators are disallowed within constant expressions
ecc: error: [9:9] in static initialization: assignment operators are disallowed within constant expressions
ecc: error: [10:10] in static initialization: increment operators are disallowed within constant expressions
ecc: error: [13:9] in static initialization: assignment operators are disallowed within constant expressions
ecc: error: [14:10] in static initialization: decrement operators are disallowed within constant expressions
ecc: error: [20:11] in static initialization: function calls are disallowed within constant expressions
ecc: error: [23:10] in static initialization: the comma operator is disallowed within constant expressions
